### PR TITLE
Port over auth, user, and courseware apps

### LINKED ISSUE #2: Account creation
As a user I would like to create an account on mitx online.
- Bring over the best of our implementation from xpro , etc. 
- Consolidate `courseware` app into the `users` app:
> `courseware` as a name is a holdover from when we envisioned xPro supporting alternative backends other than `openedx` in the future, so it served as an abstraction layer between users and arbitrary courseware backends. It's highly unlikely we'd ever do it for `mitx-online`, so we should just remove the abstraction (and any confusion caused by the name)

#### Designs and Mockups

N/A - should be barebones / wireframe style.

#### Acceptance Criteria:

- [ ] Learner can create an account
- [ ] Learner can login

#### Out of Scope

- Export compliance checks
- Entering profile data

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #2 

#### What's this PR do?
This PR primarily adds registration and login functionality, although a few other pieces of user-centric functionality was also brought in for the sake of keeping the changes minimal. The source for this implementation is mainly xPro's authentication, user, and courseware applications. I've removed as much as i reasonably can while balancing getting this completed in a timely manner.

#### How should this be manually tested?
- Configure OpenedX and mitX Online to work with each other per `docs/configure_openedx.md`. If you already have configuration in place for xPro in openedx, you'll need to alter those to point to mitx online instead, since the social auth plugin [doesn't work for multiple apps](https://github.com/mitodl/social-auth-mitxpro/issues/14), although this might also be an openedx limitation

- You can register as a new user (http://mitxonline.odl.local:8013/create-account/)
- You can logout
- You can login (http://mitxonline.odl.local:8013/signin/)

### CODE REVIEW

**Comment by rhysyngsun** [comment]
@gsidebo I think this should be all set to review now

**Comment by gsidebo** [comment]
Just verified that the auth flows all work. The setup documentation needs to be changed in a bunch of ways, but we can tackle that in a different issue

File: docker-compose-notebook.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ x-environment:
   DEV_ENV: 'True'  # necessary to have nginx connect to web container
   SECRET_KEY: local_unsafe_key
   DATABASE_URL: postgres://postgres@db:5432/postgres
+  MITX_ONLINE_BASE_URL: ${MITXPRO_BASE_URL:-http://mitxonline.odl.local:8013}
```
Comment (gsidebo) [inline docker-compose-notebook.yml:10]:
```suggestion
  MITX_ONLINE_BASE_URL: ${MITX_ONLINE_BASE_URL:-http://mitxonline.odl.local:8013}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+          sudo sysctl -w vm.max_map_count=262144
+
+      - name: Runs Elasticsearch
+        uses: elastic/elastic-github-actions/elasticsearch@master
+        with:
+          stack-version: 6.7.1
+
+      - name: Running Celery
+        run: |
+          celery -A mitxpro worker -B -l INFO &
```
Comment (gsidebo) [inline .github/workflows/ci.yml:83]: ~~`mitxpro`~~ `main` I think?

File: main/settings.py [hunk 3]
```diff
@@ -1,34 +1,35 @@
...
+# wildcard import boilerplate digital credentials settings
+# from mitol.digitalcredentials.settings import *  # pylint: disable=wildcard-import,unused-wildcard-import
+from mitol.common.settings.webpack import *  # pylint: disable=wildcard-import,unused-wildcard-import
+from redbeat import RedBeatScheduler
+
+from main.celery_utils import OffsettingSchedule
 from main.sentry import init_sentry
 
-VERSION = "0.0.0"
+VERSION = "0.90.1"
```
Comment (gsidebo) [inline main/settings.py:31]: Is this intentional?

**Review by gsidebo — CHANGES_REQUESTED** [review]
In addition to the comments: what was your approach to copying things over from `/static`? I see some files that aren't relevant yet and may not be used in the final version, e.g.: `static/js/image_carousel.js`, `static/js/hero.js`, `static/images/certificates/mitx-pro-logo-cer.png`, etc. If the idea is to cull these later, we should probably create some issues

**Comment by rhysyngsun** [comment]
@gsidebo all set for review again

**Review by gsidebo — APPROVED** [review]
Looks great! Feel free to merge after addressing those nitpicks. I'll open a separate issue for updating the edX configuration docs 👍

**Pull request merged**